aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -610,7 +618,26 @@ pub fn convert_changeset_and_events_cached<C: AccessPathCache>(
         })
         .collect::<Result<Vec<_>, VMStatus>>()?;
 
-    Ok((ws, events))
+    out_events.extend(events.into_iter());
+    Ok(())
+}
+
+pub fn convert_table_changeset(
+    table_changeset: TableChangeSet,
+    out_write_set: &mut WriteSetMut,
+) -> Result<(), VMStatus> {
+    for (handle, change) in table_changeset.changes {
+        for (key, value_opt) in change.entries {
+            let state_key = StateKey::table_item(handle.0, key);
+            if let Some(bytes) = value_opt {
+                out_write_set.push((state_key, WriteOp::Value(bytes)))
```

Comment:
@CapCap  So for a table item, the DB doesn't know it's key_type nor the value_type. so basically just opaque bytes (for the handle, the key and the value). Does the indexer want it? (playing with it here: https://github.com/aptos-labs/aptos-core/commit/dd61bc8c97932970c6e1c8c2db39a787ed0faa1b)

Reply:
Ouch :-( is there a way to get or store the types for the write set?

Reply:
We’d still want it either way, but more grokable is better ;-)

Reply:
@CapCap  The indexer can in theory discover the type info when the table is first created (as a struct under an resource, because you'll be able to parse the type info out of the resource embodying resource type.), but obviously that's not too trivial.

@sitalkedia  ^ the type info is in the table itself, as all other parts of the system: typing info is in the modules,  separate from the instance data.

Reply:
that actually sounds pretty good :-) can have a PG table for those table types :-)